File: src/server/State.py
class State:
    """The implementation of a exemplary state class.
    """

    def __init__(self):
        self._id = 0
        self._name = ""

    def get_id(self):
        """Reads out the id."""
        return self._id

    def set_id(self, val):
        """Sets the id."""
        self._id = val
    
    def get_name(self):
        """Reads out the name."""
        return self._name

    def set_name(self, txt):
        """Sets the name."""
        self._name = txt

    @staticmethod
    def from_dict(dicti=dict()):

        """"Converts a Python dict() into a State()."""
        state = State()
        state.set_id(dicti["id"])
        state.set_name(dicti["name"])
        return state

File: frontend/src/api/State.js


export default class State {

    /// Represents a state 
    
        constructor(aId, aName){
            
            this.id=aId;
            this.name=aName;
        }
    
        //Sets the id
        setID(aID){
          this.id = aID;
        }
    
        //Reads out the id
        getID(){
            return this.id;
        }

        //Sets the name
        setName(aName){
            this.name = aName;
          }
      
        //Reads out the name
        getName(){
            return this.name;
        }
    
        //Returns an Array of States from a given JSON structure.
    
        static fromJSON(states) {
            let result = [];
    
            if (Array.isArray(states)) {
                states.forEach((s) => {
                    Object.setPrototypeOf(s, State.prototype);
                    result.push(s);
                })
            }
            // it's a single object and not an array
            else {
                let s = states;
                Object.setPrototypeOf(s, State.prototype);
                result.push(s);
            }
    
            return result;
        }
    }

    